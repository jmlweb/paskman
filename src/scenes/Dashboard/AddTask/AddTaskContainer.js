// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { modalToggle } from '../../../components/Modal/duck';
import AddTask from './AddTask';

class AddTaskContainer extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return <AddTask />;
  }
}

export function mapStateToProps(state: {
  components: {
    modal: mixed,
  }
}) {
  return { ...state.components.modal };
};

export const mapDispatchToProps = {
  modalToggle,
};

const AddTaskConnectedContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(AddTaskContainer);

export default AddTaskConnectedContainer;
