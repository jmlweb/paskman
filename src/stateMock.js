const stateMock = Object.freeze({
  app: {
    resizer: {
      dimensions: {
        width: 320,
        height: 568,
      },
    },
    topBar: {
      menuOpen: false,
    },
  },
  components: {
    modal: {},
  },
  data: {
    settings: {
      isFetching: false,
      isSaving: false,
      target: {
        working: 25,
        resting: 5,
      },
      pauseBetween: false,
      confirmEndingTask: true,
    },
  },
});

export default stateMock;
