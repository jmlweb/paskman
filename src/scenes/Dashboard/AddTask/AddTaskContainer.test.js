import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import AddTaskContainer from './AddTaskContainer';
import stateMock from '../../../stateMock';

describe('AddTaskContainer', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  let wrapper;
  let store;
  const modalToggle = jest.fn();
  beforeEach(() => {
    store = mockStore(stateMock);
    wrapper = mount(
      <Provider store={store}>
        <AddTaskContainer modalToggle={modalToggle} />
      </Provider>
    );
  });
  it('Render the component', () => {
    expect(wrapper).toBeDefined();
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
