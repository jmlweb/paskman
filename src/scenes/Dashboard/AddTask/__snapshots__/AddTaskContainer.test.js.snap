// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskContainer Matchs snapshot 1`] = `
<div
  className="sc-htpNat iYGpdw"
>
  <div
    className="sc-bxivhb iRWXmO"
  >
    <h1
      className="sc-ifAKCX ckbHNo sc-ckVGcZ kliXPK"
    >
      Add new task
    </h1>
    <button
      className="sc-bwzfXH cyuhjn"
      onClick={[Function]}
    >
      <svg
        height="35"
        viewBox="0 0 35 35"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#1E2F76"
          fillRule="nonzero"
        >
          <path
            d="M17.5 35C7.85 35 0 27.15 0 17.5S7.85 0 17.5 0 35 7.85 35 17.5 27.15 35 17.5 35zm0-32.813c-8.444 0-15.313 6.87-15.313 15.313 0 8.444 6.87 15.313 15.313 15.313 8.444 0 15.313-6.87 15.313-15.313 0-8.444-6.87-15.313-15.313-15.313z"
          />
          <path
            d="M22.97 24.063c-.28 0-.56-.108-.775-.32L11.258 12.804c-.428-.428-.428-1.12 0-1.547.428-.428 1.12-.428 1.547 0l10.937 10.937c.428.428.428 1.12 0 1.547-.213.213-.493.32-.773.32z"
          />
          <path
            d="M22.97 24.063c-.28 0-.56-.108-.775-.32L11.258 12.804c-.428-.428-.428-1.12 0-1.547.428-.428 1.12-.428 1.547 0l10.937 10.937c.428.428.428 1.12 0 1.547-.213.213-.493.32-.773.32z"
          />
          <path
            d="M12.03 24.063c-.28 0-.56-.108-.772-.32-.428-.43-.428-1.12 0-1.548l10.937-10.937c.428-.428 1.12-.428 1.547 0 .428.428.428 1.12 0 1.547L12.805 23.742c-.214.213-.494.32-.774.32z"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    className="sc-EHOje lixEVO"
  >
    <form
      className="sc-bZQynM hSufOU"
    >
      <fieldset
        className="sc-gzVnrw cJCIgP"
      >
        <div
          className="sc-htoDjs hpWqMX"
        >
          <label
            className="sc-dnqmqq jebcPH"
          >
            Name (3 chars min)
          </label>
          <input
            className="sc-iwsKbI WHWzN"
            id="name"
            minLength="3"
            placeholder="Try to be concise"
            required={true}
            type="text"
          />
        </div>
        <div
          className="sc-htoDjs hpWqMX"
        >
          <label
            className="sc-dnqmqq jebcPH"
          >
            Description
          </label>
          <textarea
            className="sc-gZMcBi jEvzYf"
            cols="80"
            id="description"
            placeholder="Describe the task"
            rows="4"
          />
        </div>
        <div
          className="sc-htoDjs hpWqMX"
        >
          <label
            className="sc-dnqmqq jebcPH"
          >
            Pomodoros required
          </label>
          <div
            className="sc-gqjmRU hpLqBK"
          >
            <label
              className="sc-VigVT dzzzHZ"
              selected={true}
            >
              <input
                className="sc-jTzLTM hbqTmN"
                id="id-25"
                onClick={[Function]}
                type="radio"
                value={25}
              />
              <span>
                1
              </span>
            </label>
            <label
              className="sc-VigVT cnzhie"
              selected={false}
            >
              <input
                className="sc-jTzLTM hbqTmN"
                id="id-50"
                onClick={[Function]}
                type="radio"
                value={50}
              />
              <span>
                2
              </span>
            </label>
            <label
              className="sc-VigVT cnzhie"
              selected={false}
            >
              <input
                className="sc-jTzLTM hbqTmN"
                id="id-75"
                onClick={[Function]}
                type="radio"
                value={75}
              />
              <span>
                3
              </span>
            </label>
            <label
              className="sc-VigVT cnzhie"
              selected={false}
            >
              <input
                className="sc-jTzLTM hbqTmN"
                id="id-100"
                onClick={[Function]}
                type="radio"
                value={100}
              />
              <span>
                4
              </span>
            </label>
            <label
              className="sc-VigVT cnzhie"
              selected={false}
            >
              <input
                className="sc-jTzLTM hbqTmN"
                id="id-125"
                onClick={[Function]}
                type="radio"
                value={125}
              />
              <span>
                5
              </span>
            </label>
          </div>
          <div
            className="sc-jzJRlG bzLuwb"
          >
            <button
              className="sc-jKJlTe cHiUNk"
              color="primary"
              size="sm"
            >
              Change to time
            </button>
          </div>
        </div>
      </fieldset>
      <button
        className="sc-eNQAEJ iaXIpL"
        color="success"
        size="md"
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
